sofascoreAcademy/sofascoreAcademy/ViewModel/EventsViewModel.swift

```diff
@@ -7,14 +7,34 @@
 
 import Foundation
 import UIKit
-import SofaAcademic
 import SnapKit
 
-class EventsViewModel {
+class EventsViewModel: ObservableObject {
+    enum State {
+        case idle
+        case loaded([LeagueSection])
+        case loading
+        case error
+    }
+    @Published private(set) var state: State = .idle
+    
     var sections: [LeagueSection] = []
     func loadSections() async {
-        let events = Homework3DataSource().events()
-        sections = getLeagueSections(for: events)
+       
+        self.state = .loading
+        do {
+            let events = try await APIClient.getEvents(sport: MenuViewModel.shared.selectedSport.urlSlug)
```

Comment:
Ako želiš imati shared state za nešto, to nebi trebao biti neki ViewModel od nekog viewa. To bi trebao biti neki globalni service/helper/manager, npr. `SprotSelectionService`.